lib/Analysis/UpdateGraphOutputSize.cpp

```diff
@@ -120,53 +25,6 @@ static unsigned int getBatchSize()
   return batch_size;
 }
 
-//===----------------------------------------------------------------------===//
-// UpdateGraphOutputSize
-//===----------------------------------------------------------------------===//
-
-bool UpdateGraphOutputSize::updateReshapeOutputInfo(xGraph *pGraph)
-{
-  bool require_onnxInferShape = false;
-  for (xNode *node : pGraph->nodes()) {
-    const auto kind = node->kind();
-    if (kind != xBuiltinSymbol::kReshape)
-      continue;
-
-    // check special case, the output dimension(shape) can be 0 or -1
-    // At most one dimension of the new shape can be -1. In this case, the
-    // value is inferred from the size of the tensor and the remaining
-    // dimensions. A dimension could also be 0, in which case the actual
-    // dimension value is unchanged (i.e. taken from the input tensor).
-    // this case require input tensor dimension value
-    std::vector<int64_t> dest_dims = getReshapeOpDestDims(node);
-    if (std::find(dest_dims.begin(), dest_dims.end(), 0) != dest_dims.end() or
-        std::find(dest_dims.begin(), dest_dims.end(), -1) != dest_dims.end()) {
-      xValue *input0 = node->inputs()[0];
-      if (input0->sizes().size() == 0) {
-        // input tensor dimension is empty, need to run onnx shape inference
-        // first
-        return true;
-      }
-      // update shape tensor with exact dimension value
-      updateReshapeOpDestDimension(pGraph, node, dest_dims);
-      require_onnxInferShape = true;
-    }
-
-    // user specific batch_size
-    if (m_BatchSize > 0)
-      dest_dims[0] = m_BatchSize;
```

Comment:
NOTE: this patch will disable dynamic batch size feature,
maybe you need disable this feature by remove batch_size option.

Reply:
Oh god! I will fix it next week.
I think we can still keep this feature by reorder it?
(modify all node's batch size after onnx shape inference?)

Reply:
yes, it's good solution!